biorxiv/pca_association_experiment/biorxiv_pca_category_bootstrap.py

```diff
@@ -0,0 +1,230 @@
+)
+journal_map_df.head()
+
+
+# # PCA the Documents
+
+# Run PCA over the documents. Generates 50 principal components, but can generate more or less.
+
+# In[3]:
+
+
+n_components = 50
+random_state = 100
+
+
+# In[4]:
+
+
+biorxiv_articles_df = pd.read_csv(
+    Path("..")/
+    Path("word_vector_experiment")/
+    Path("output/word2vec_output")/
+    Path("biorxiv_all_articles_300.tsv.xz"), 
+    sep="\t"
+)
+biorxiv_articles_df.head()
+
+
+# In[5]:
+
+
+reducer = PCA(
+    n_components = n_components,
+    random_state = random_state
+)
+
+reducer.fit(
+    biorxiv_articles_df[[f"feat_{idx}" for idx in range(300)]]
+    .values
+)
+
+
+# # Bootstrap 95% Confidence Intervals Cosine sim
+
+# In[6]:
+
+
+document_categories_df = (
+    journal_map_df[["document", "category"]]
+    .merge(biorxiv_articles_df, on="document")
+)
+document_categories_df.head()
+
+
+# In[7]:
+
+
+boostraped_iterations = 10000
+pc_1 = []
+pc_2 = []
+
+for iteration in tqdm_notebook(range(boostraped_iterations)):
+    sampled_df = (
+        document_categories_df
+        .groupby("category")
+        .apply(lambda x: x.sample(1, random_state=iteration))
+        .reset_index(drop=True)
+        .sort_values("category")
+    )
+    
+    document_distance = (
+        1 - cdist(
+            sampled_df
+            .drop(["document", "category"], axis=1)
+            .values, 
+            reducer.components_[0:2], 
+            'cosine'
+        )
+    )
+    
+    pc_1.append(document_distance[:, 0])
+    pc_2.append(document_distance[:, 1])
+
+pc_1 = np.stack(pc_1)
+pc_2 = np.stack(pc_2)
+
+
+# In[8]:
+
+
+mean_similarity_df = (
+    pd.DataFrame(
+        (
+            1 - cdist(
+                document_categories_df
+                .drop(["document", "category"], axis=1)
+                .values, 
+                reducer.components_[0:2], 
+                'cosine'
+            )
+        ),
+        columns=["pca1_cossim", "pca2_cossim"]
+    )
+    .assign(
+        category = document_categories_df.category.tolist(),
+        document = document_categories_df.document.tolist()
+    )
+)
+mean_similarity_df.head()
+
+
+# In[9]:
+
+
+category_sim_df = (
+    mean_similarity_df
+    .groupby("category")
+    .agg({
+        "pca1_cossim": "mean",
+        "pca2_cossim": "mean"
+    })
+    .assign(
+        pca1_cossim_upper = np.percentile(pc_1, 97.5, axis=0),
+        pca1_cossim_lower = np.percentile(pc_1, 25, axis=0),
+        pca2_cossim_upper = np.percentile(pc_2, 97.5, axis=0),
+        pca2_cossim_lower = np.percentile(pc_2, 25, axis=0)
+    )
+    .reset_index()
+    [
+        [
+            "category", "pca1_cossim_lower",
+            "pca1_cossim", "pca1_cossim_upper",
+            "pca2_cossim_lower", "pca2_cossim",
+            "pca2_cossim_upper"
+        ]
+    ]
+)
+category_sim_df.head()
+
+
+# In[10]:
+
+
+category_sim_df.to_csv(
+    "output/category_cossim_95_ci.tsv", 
+    sep="\t", index=False
+)
+
+
+# In[14]:
+
+
+g = (
```

Comment:
So these are showing the variability of docs within a given category. Looks pretty variable..is this meant to give context for your journal predictions? So that you are able to say predict X well because of the variability of documents in category Y?

Reply:
> is this meant to give context for your journal predictions? So that you are able to say predict X well because of the variability of documents in category Y?

No. This is more along the lines of highlighting how categories are related with a given principal component. The idea here is to give the reader a sense of the type of research is being discussed in various preprints. For example the biochemistry category has a mean at 0 and a large variance for the PC1. PC1 is about molecular biology vs quantitative biology. This means on average biochemistry preprints contain an equal amount of molecular and quantitative experiments; however, a greater number of preprints contain quantitative biology as shown by the positive skew of the variance.

Reply:
I see! I wonder if it would help to change the axis label to `similarity of document vs PC1` and maybe explain what PCs you're using in this case if PC1 = biochemistry in this case